v2/tools/asoctl/internal/crd/cleaner.go

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package crd
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	apiextensionsV1 "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func CleanDeprecatedCRDVersions(ctx context.Context, cl apiextensionsV1.CustomResourceDefinitionInterface) error {
+	var crdRegexp = regexp.MustCompile(".*\\.azure\\.com")
+	var deprecatedVersionRegexp = regexp.MustCompile("v1alpha1api.{8}.*")
```

Comment:
```suggestion
	var deprecatedVersionRegexp = regexp.MustCompile("v1alpha1api\d{8}(preview)?(storage)?")
```

Is a more precise match, I think?